### ISSUE TITLE
Silently fails with invalid template

### ISSUE THREAD

**Issue by jsumners** [jhiver/template-tal#1]
Given the template

``` html
<!doctype html>
<html>
<head>
  <meta charset="utf-8">
  <title tal:content="self.title">a title</title>
</head>
<body>
  <p>stuff</p>
</body>
</html>
```

`tal.process()` will fail with no feedback. Ideally, `tal.process` would return a `Promise` that issues a `reject` upon failure. A Node style `callback(err, result)` would also suffice.

**Comment by jhiver** [comment]
Thanks for this, I'll take a look...

**Comment by jhiver** [comment]
Hello again. It doesn't fail silently, it throws an error message. Why would you return a Promise or use a callback for an operation which is not asynchronous?

**Comment by jsumners** [comment]
I'm sorry, I don't even remember what I was trying to use this project in. But if it `throw`s an error, that should be documented. `try...catch` isn't a common pattern in JavaScript.

**Comment by jhiver** [comment]
better late than never... fixed in 0.2.2 among a bunch of other things.
